### ISSUE TITLE
Stricter validation for CLI options and syntax

### ISSUE THREAD

**Issue by unsystemizer** [bitcoin/bitcoin#11819]
Using 0.15.1 from BitcoinCore.org:
```
$ bitcoin-qt -rescan 1 -wallet otherWallet.dat 
PaymentServer::ipcSendCommandLine: Payment request file does not exist:  "1"
"PaymentServer::verifySize: Payment request too large (720896 bytes, allowed 50000 bytes)."
```
Maybe this pilot error could be handled better. For example, Bitcoin-Qt (or bitcoind) could refuse to start.

Note that in the above case Bitcoin-Qt silently ends up using a wrong (default) wallet.

**Comment by promag** [comment]
> Note that in the above case Bitcoin-Qt silently ends up using a wrong (default) wallet.

It should be `-wallet=...`.

**Comment by unsystemizer** [comment]
I know it’s all wrong, but the problem is despite the both options being syntactically wrong, service still starts, whereas it IMO shouldn’t. 

Thanks for linking to the existing issue. I disagree with the [view](https://github.com/#issuecomment-4991036) that it’s worse to fail to start than start with a wrong wallet or with “fake rescan”(or whatever else is passed and appears to work whereas it doesn’t at all). 

The above example illustrates how a fake rescan of a wrong (new) wallet results in no funds found, ie after your daemon starts you conclude that your old otherWallet.dat can be deleted because it’s empty.

Why put convenience before correctness when it comes to daemon startup options?
No one who didn’t mess with their settings will have a problem starting service, and those who did shouldn’t succeed to start it if they passed syntactically wrong options. 

I am okay if the maintainers close this because it’s a duplicate.

**Comment by promag** [comment]
I think it's fine to not launch the process in these cases.

**Comment by Sjors** [comment]
Concept ACK on more aggressively checking parameters. Also agree that this ticket can be closed as duplicate of #1044.
